Tests/Apollo/Network/HttpNetworkTransportTests.swift

```diff
@@ -0,0 +1,342 @@
+    XCTAssert(request!.url?.host == network.url.host)
+    XCTAssert(request!.httpMethod == "POST")
+    
+    validatePostBody(with: request!,
+                     query: query,
+                     queryDocument: true)
+  }
+  
+  func testNotUseGETForQueriesAPQsRequest() {
+    let network = HTTPNetworkTransport(url: URL(string: endpoint)!,
+                                       useGETForQueries: false,
+                                       enableAutoPersistedQueries: true,
+                                       useHttpGetMethodForPersistedQueries: false)
+    let mockSession = MockURLSession()
+    network.session = mockSession
+    let query = HeroNameQuery(episode: .empire)
+    let _ = network.send(operation: query) { _,_ in }
+    let request = mockSession.lastRequest
+    
+    XCTAssertNotNil(request)
+    XCTAssert(request!.url?.host == network.url.host)
+    XCTAssert(request!.httpMethod == "POST")
+    
+    validatePostBody(with: request!,
+                     query: query,
+                     queryDocument: false,
+                     persistedQuery: true,
+                     variable: true)
+  }
+  
+  func testUseGETForQueriesAPQsRequest() {
+    let network = HTTPNetworkTransport(url: URL(string: endpoint)!,
+                                       useGETForQueries: true,
+                                       enableAutoPersistedQueries: true,
+                                       useHttpGetMethodForPersistedQueries: false)
+    let mockSession = MockURLSession()
+    network.session = mockSession
+    let query = HeroNameQuery(episode: .empire)
+    let _ = network.send(operation: query) { _,_ in }
+    let request = mockSession.lastRequest
+    
+    XCTAssertNotNil(request)
+    XCTAssert(request!.url?.host == network.url.host)
+    XCTAssert(request!.httpMethod == "GET")
+    
+    validateUrlParams(with: request!,
+                      query: query,
+                      queryDocument: false,
+                      persistedQuery: true,
+                      variable: true)
+  }
+  
+  func testNotUseGETForQueriesAPQsGETRequest() {
+    let network = HTTPNetworkTransport(url: URL(string: endpoint)!,
+                                       useGETForQueries: false,
+                                       enableAutoPersistedQueries: true,
+                                       useHttpGetMethodForPersistedQueries: true)
+    let mockSession = MockURLSession()
+    network.session = mockSession
+    let query = HeroNameQuery(episode: .empire)
+    let _ = network.send(operation: query) { _,_ in }
+    let request = mockSession.lastRequest
+    
+    XCTAssertNotNil(request)
+    XCTAssert(request!.url?.host == network.url.host)
+    XCTAssert(request!.httpMethod == "GET")
+    
+    validateUrlParams(with: request!,
+                      query: query,
+                      queryDocument: false,
+                      persistedQuery: true,
+                      variable: true)
+  }
+}
+
+// Mark: Helpers
```

Comment:
The `// MARK:` directive doesn't really do anything in Swift unless you use all uppercase and a semicolon.

Reply:
updated